pkg/filepiece/fpiece.go

```diff
@@ -0,0 +1,143 @@
+// Copyright (C) 2018 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package fpiece
+
+import (
+	"errors"
+	"io"
+	"os"
+)
+
+// Chunk - Section of data to be concurrently read
+type Chunk struct {
+	file       *os.File
+	start      int64
+	final      int64
+	currentPos int64
+}
+
+// seek enum
+const (
+	SeekStart   = 0 // seek relative to the origin of the file
+	SeekCurrent = 1 // seek relative to the current offset
+	SeekEnd     = 2 // seek relative to the end
+)
```

Comment:
Oh, why don't you try the super wierd `iota` identifier? https://stackoverflow.com/questions/14426366/what-is-an-idiomatic-way-of-representing-enums-in-go#14426447